.scalafmt.conf

```diff
@@ -1,51 +1,93 @@
-version = 3.9.4
+# Main goals:
+# - nicer commit diffs (trailing commas, no alignment for pattern matching, force new lines)
+# - better interop with default IntelliJ IDEA setup (matching import and modifiers sorting logic)
+# - better developer experience on laptop screens (like 16' MBPs) with IntelliJ IDEA (line wraps)
 
-project.git = true
+version = 3.9.4
 
-maxColumn = 130
+runner.dialect = scala213source3
 
+maxColumn = 120
 trailingCommas = always
 
-continuationIndent {
-  callSite = 2
+preset = default
+# do not align to make nicer commit diffs
+align.preset = none
+
+indent {
+  # altering defnSite and extendSite to have this:
+  #   final class MyErr extends RuntimeException(
+  #     "super error message",
+  #   )
+  # instead of this:
+  #   final class MyErr extends RuntimeException(
+  #       "super error message",
+  #     )
   defnSite = 2
+  extendSite = 0
 }
 
-align.preset = some
-align.tokenCategory {
-  LeftArrow = Assign
-  Equals = Assign
+spaces {
+  # makes string interpolation with curlies more visually distinct
+  inInterpolatedStringCurlyBraces = true
+}
+
+newlines {
+  # keep author new lines where possible
+  source = keep
+  # force new line after "(implicit" for multi-line arg lists
+  implicitParamListModifierForce = [after]
+  avoidForSimpleOverflow = [
+    tooLong, # if the line would be too long even after newline inserted, do nothing
+    slc, # do nothing if overflow caused by single line comment
+  ]
 }
-# Mostly subset of `align.preset=more`, but with extra settings for `=`
-# For all settings see `AlignToken#default` in
-# https://github.com/scalameta/scalafmt/blob/master/scalafmt-core/shared/src/main/scala/org/scalafmt/config/AlignToken.scala
-align.tokens."+" = [
-  { code = "%", owner = "Term.ApplyInfix" }
-  { code = "%%", owner = "Term.ApplyInfix" }
-  { code = "%%%", owner = "Term.ApplyInfix" }
-  { code = "=>", owner = "(Case|Term.Function)" }
-  { code = "<-", owner = "Enumerator.Generator" }
-  { code = "=", owner = "(Defn.Val|Defn.Var|Type|Def|Enumerator.Val|Assign|Term.Param)" } # Defn.Val is mostly for Dependencies.scala,
-  { code = "->", owner = "Term.ApplyInfix" }
-]
 
-newlines.implicitParamListModifierPrefer = before
-newlines.beforeCurlyLambdaParams = multilineWithCaseOnly
+verticalMultiline {
+  atDefnSite = true
+  arityThreshold = 3 # more than 3 args in a list will be turned vertical
+  newlineAfterOpenParen = true # for nicer commit diffs
+}
 
-indentOperator.topLevelOnly = true
+# for nicer commit diffs - forces new line before last parenthesis:
+#   class MyCls(
+#     arg1: String,
+#     arg2: String,
+#   ) extends MyTrait {
+#
+# without it:
+#   class MyCls(
+#     arg1: String,
+#     arg2: String) extends MyTrait {
+danglingParentheses.exclude = []
 
-docstrings.style = keep
```

Comment:
this was added to "force-keep" the formatting of some ScalaDocs, where lists are used - without it ScalaDoc reformats list to paragraph

Reply:
I manually reviewed all scaladoc format changes in files and fixed a couple of places where it broke the markup.